compiler/src/main/kotlin/ast/reduce.kt

```diff
@@ -204,12 +240,22 @@ private fun reduceDecl(node: ParseTree): Decl {
     }
 }
 
+/**
+ * Transform all statements in a code block into abstract syntax tree nodes.
+ *
+ * @throws AssertionError thrown when encountering an unexpected node in the parse tree.
+ */
 fun reduceCodeBlock(block: CellmataParser.Code_blockContext): List<Stmt> {
     return block.children
         .filter { it !is TerminalNode } // Remove terminals
         .map { reduceStmt(it) }
 }
 
+/**
+ * Attempts to recursively transform an ANTLR parse tree to an abstract syntax tree.
```

Comment:
"Attempts to recursively transform an ANTLR parse tree to an abstract syntax tree." has been repeated thrice now, without mentioning that it reduces first a `stmt`, then a `decl`, lastly here the start-symbol.

I suggest that it is removed or elaborated.

Reply:
> Attempts to recursively transform an declaration node in the parse tree to create a subtree of the AST.

Would this be an acceptable phrasing for stmt, decl, and expr?

> Attempts to recursively transform an declaration node in the parse tree to create a subtree of the AST.

And how is this for the main `reduce` function?

Reply:
Yes, but use the stmt, decl, and expr where appropriate, e.g.;

For expression:
> Attempts to recursively transform an expression node in the parse tree to create a subtree of the AST.

I think you made a minor mistake with copy/paste in your reply, too.